scripts/update-changelog.mjs

```diff
@@ -0,0 +1,83 @@
+import { createRequire } from 'node:module';
+import { exec } from 'node:child_process';
+import fs from 'node:fs';
+import process from 'node:process';
+import util from 'node:util';
+
+import { Release, parser } from 'keep-a-changelog';
+import increment from 'semver/functions/inc.js';
+import read from '@changesets/read';
+
+const require = createRequire(import.meta.url);
+const pkg = require('../package.json');
+const changesets = await read('./');
+const groups = { security: [], removed: [], changed: [], deprecated: [], added: [], fixed: [] };
+const execAsync = util.promisify(exec);
+
+let releaseType = 'patch';
+
+for (const {
+	releases: [{ type }],
+	summary,
+	id,
+} of changesets) {
+	const [change, item] = summary.split(/: (.+)/s);
+
+	let stdout;
+
+	try {
+		({ stdout } = await execAsync(`git log --pretty=format:"%ae %s" .changeset/${id}.md`));
+	} catch (e) {
+		console.error(e.message);
+	}
+
+	const [email, message] = stdout.split(/ (.*)/s);
+	const [, user] = email.match(/(?:\d+\+)?([^@]+)(?=@)/);
+	const [issue] = message.match(/#[0-9]{4,}/);
+	const PR = `https://github.com/stylelint/stylelint/pull/${issue.slice(1)}`;
+	const account = `https://github.com/${user}`;
+
+	groups[change.toLowerCase()].push(`${item} ([${issue}](${PR})) ([@${user}](${account})).`);
+
+	if (releaseType !== 'major' && type !== 'patch') releaseType = type;
+}
+
+class CustomRelease extends Release {
```

Comment:
note

required because the default order or the subheaders is different from ours
see https://github.com/oscarotero/keep-a-changelog/blob/master/src/Release.ts#L24-L31